main.cpp

```diff
@@ -275,296 +300,281 @@ struct Bakery
+    void modulateOutput(float vModAtt, float modSpeed);
     void silenceOutput();
 };
 
-/*
-Thing 6) filter
-5 properties:
-    1) cutoff frequency (float)
-    2) resonance (float)
-    3) filter type (std::string)
-    4) input gain (double)
-    5) overdrive (float)
-3 things it can do:
-    1) cut low frequencies
-    2) cut high frequencies
-    3) self-resonate
- */
+double VCA::setInitGain(double vMinGain, double vMaxGain)
+{
+    std::cout << "Setting init gain. Current minimum is: " << vMinGain << " and maximum is: " << vMaxGain << std::endl;
+
+    return (vMinGain-vMaxGain);
+}
+
+void VCA::modulateOutput(float vModAtt, float modSpeed)
+{
+    std::cout << "Modulating output by: " << vModAtt << " with the frequency of: " << modSpeed << std::endl;
+}
+
+void VCA::silenceOutput()
+{
+}
 
 struct Filter
 {
-    //1) cutoff frequency (float)
     float cutoffFreq = 13200.2f;
-    //2) resonance (float)
     float resonance = 0.77f;
-    //3) filter type (std::string)
     std::string filterType = "Low Pass";
-    //4) input gain (double)
     double inputGain = -5.3;
-    //5) overdrive (float)
     float overdriveAmount = 1.7f;
 
-    //1) cut low frequencies
     void cutLows(float cutoffFreq, float resonance);
-    //2) cut high frequencies
     void cutHighs(float cutoffFreq, float resonance);
-    //3) self-resonate
-    bool selfResonate(float resonance, double inputGain);
+    bool selfResonate(float fResonance, double fInputGain);
 };
 
-/*
-Thing 7) envelope
-5 properties:
-    1) attack time (float)
-    2) decay time (float)
-    3) sustain level (double)
-    4) release time (float)
-    5) curve (float)
-3 things it can do:
-    1) create long sustained signals
-    2) create short staccato signals
-    3) loop over time
- */
+void Filter::cutLows(float fCutoffFreq, float fResonance)
+{
+    std::cout << "Cutting low frequencies below: " << fCutoffFreq << "with a Q of: " << fResonance << std::endl;
+}
+void Filter::cutHighs(float fCutoffFreq, float fResonance)
+{
+    std::cout << "Cutting high frequencies above: " << fCutoffFreq << "with a Q of: " << fResonance << std::endl;
+}
+bool Filter::selfResonate(float fResonance, double fInputGain)
+{
+    return fResonance > 0.9f && fInputGain > -35.0;
+}
 
 struct Envelope
 {
-    //1) attack time (float)
     float attackTime = 22.5f;
-    //2) decay time (float)
     float decayTime = 330.2f;
-    //3) sustain level (double)
     double sustainLevel = 0.99228;
-    //4) release time (float)
     float releaseTime = 675.1f;
-    //5) curve (float)
     float curve = 0.9f;
 
-    //1) create long sustained signals
-    void padPreset(float curve);
-    //2) create short staccato signals
-    void staccatoPreset(float curve);
-    //3) loop over time
-    bool loopOver(float attackTime, float decayTime, float releaseTime);
+    void padPreset(float eCurve);
+    void staccatoPreset(float eCurve);
+    bool loopOver(float eAttackTime, float eDecayTime, float eReleaseTime);
```

Comment:
Review our Slack discussion of using member variables in member functions directly. 
Change your argument list so that your function arguments are not duplicates of your member variables.
This can mean changing the function name (meaning the function purpose has changed) 
or it could mean changing the function arguments.
Remember, name things what they are or what they do.
- [ ] revise